src/lib/y2packager/dialogs/addon_selector.rb

```diff
@@ -119,14 +137,51 @@ def dialog_content
         )
       end
 
-      def read_user_selection
-        selected_items = Yast::UI.QueryWidget(Id("addon_repos"), :SelectedItems)
+      # select the dependant products for the active selection
+      def select_dependant_products
+        # select the dependent products
+        new_selection = current_selection
+
+        # the selection has not changed, nothing to do
+        return if new_selection == selected_products
+
+        # add the dependant items to the selected list
+        selected_items = Yast::UI.QueryWidget(Id(:addon_repos), :SelectedItems)
+        new_items = new_selection - selected_products
+        new_items.each do |p|
```

Comment:
NP: you can write it as 
```
selected_items = new_items.each_with_object(Yast::UI.QueryWidget(Id(:addon_repos), :SelectedItems)) do |p, res|
  res.concat(p.depends_on)
end
```

Reply:
I'd keep the current code, this uses one too long line which would need to be split and decreasing the readability...